src/peer_manager.rs

```diff
@@ -391,9 +392,32 @@ impl PeerManager {
         (ids_to_drop, our_new_prefix)
     }
 
+    /// Checks whether we have a quorum of nodes in each section
+    fn is_merging_possible(&self) -> bool {
+        let prefixes = self.expected_peers
+            .keys()
+            .map(|x| self.routing_table.find_group_prefix(x))
+            .collect::<HashSet<_>>();
+        if prefixes.contains(&None) {
+            // we expect contacts that don't belong in any of the sections in our RT - so we have
+            // no contacts from their section
+            return false;
```

Comment:
Isn't it possible when we for example just merged and got notified of new neighbouring sections that we should connect to, but haven't connected to anyone from them yet?